pkg/action/env.container.go

```diff
@@ -62,7 +64,7 @@ func (c *containerEnv) Execute(ctx context.Context, a *Action) (err error) {
 	actConf := a.ActionDef()
 	log.Debug("Starting execution of the action %q in %q environment, command %v", a.ID, c.dtype, actConf.Command)
 	// @todo consider reusing the same container and run exec
-	name := genContainerName(a, c.prefix, nil)
+	name := genContainerName(a, c.prefix, c.suffix, nil)
 	existing := c.driver.ContainerList(ctx, types.ContainerListOptions{SearchName: name})
```

Comment:
Here we check if the container already exists. We already have a randomized name, just return an error, that the container couldn't be created if something was found. The issue may appear rarely. Then we don't need `exMap` and related functionality.